src/olympia/reviewers/tests/test_views.py

```diff
@@ -3828,11 +3801,35 @@ def test_cant_contentreview_if_admin_content_review_flag_is_set(self):
             'action': 'confirm_auto_approved',
             'comments': 'ignore me this action does not support comments'
         })
-        # The request will succeed but nothing will happen.
-        assert response.status_code == 200
+        assert response.status_code == 200  # Form error
         assert ActivityLog.objects.filter(
             action=amo.LOG.APPROVE_CONTENT.id).count() == 0
 
+    def test_can_contentreview_if_addon_has_sources_attached(self):
+        GroupUser.objects.filter(user=self.reviewer).all().delete()
+        self.url = reverse(
+            'reviewers.review', args=['content', self.addon.slug])
+        summary = AutoApprovalSummary.objects.create(
+            version=self.addon.current_version, verdict=amo.AUTO_APPROVED)
+        self.addon.current_version.update(source='/path/to/fake/file.zip')
+        self.grant_permission(self.reviewer, 'Addons:ContentReview')
+        response = self.client.post(self.url, {
+            'action': 'confirm_auto_approved',
+            'comments': 'ignore me this action does not support comments'
+        })
+        assert response.status_code == 302
+        summary.reload()
+        assert summary.confirmed is None  # We're only doing a content review.
+        assert ActivityLog.objects.filter(
+            action=amo.LOG.CONFIRM_AUTO_APPROVED.id).count() == 0
+        assert ActivityLog.objects.filter(
+            action=amo.LOG.APPROVE_CONTENT.id).count() == 1
+        a_log = ActivityLog.objects.filter(
+            action=amo.LOG.APPROVE_CONTENT.id).get()
+        assert a_log.details['version'] == self.addon.current_version.version
+        assert a_log.details['comments'] == ''
+        self.assert3xx(response, reverse('reviewers.queue_content_review'))
+
```

Comment:
This isn't a replacement for the test you removed as it doesn't set the `AddonReviewerFlags` - as it just sets the source it would have passed anyway.

Reply:
I believe this is a valid replacement. The test I removed was passing either way, because it was testing the `needs_admin_code_review`. However, we decided that the deciding factor should be `version.source`, which typically comes with `needs_admin_code_review` but this also covers the case where an add-on was restricted to admin review for a different reason, in which a regular reviewer should not be allowed to review the add-on.

Reply:
You are right, I'll add it back.

Reply:
shouldn't you be testing the exact case set-up in the code below - i.e. an add-on that has `AddonReviewFlags.needs_admin_code_review` and has `self.version.source` would be allowed a content review?  The test doesn't add that flag first.